components/omega/src/ocn/CustomTendencyTerms.h

```diff
@@ -0,0 +1,181 @@
+
+namespace OMEGA {
+
+//===-----------------------------------------------------------------------===/
+// A class for the manufactured solution tendency terms
+//===-----------------------------------------------------------------------===/
+class ManufacturedSolution {
+
+   //===--------------------------------------------------------------------===/
+   // Manufactured tendency term for the thickness equation
+   //===--------------------------------------------------------------------===/
+   struct ManufacturedThicknessTendency {
+
+      // Constants defined in 'init'
+      TimeInstant ReferenceTime;
+      R8 H0;
+      R8 Eta0;
+      R8 Kx;
+      R8 Ky;
+      R8 AngFreq;
+
+      void operator()(Array2DReal ThicknessTend, const OceanState *State,
+                      const AuxiliaryState *AuxState, int ThickTimeLevel,
+                      int VelTimeLevel, TimeInstant Time) {
+
+         // Get elapsed time since reference time
+         R8 ElapsedTimeSec;
+         TimeInterval ElapsedTimeInterval = Time - ReferenceTime;
+         ElapsedTimeInterval.get(ElapsedTimeSec, TimeUnits::Seconds);
+
+         auto *Mesh       = HorzMesh::getDefault();
+         auto NVertLevels = ThicknessTend.extent_int(1);
+
+         Array1DReal XCell = Mesh->XCell;
+         Array1DReal YCell = Mesh->YCell;
+
+         OMEGA_SCOPE(LocXCell, XCell);
+         OMEGA_SCOPE(LocYCell, YCell);
+         OMEGA_SCOPE(LocH0, H0);
+         OMEGA_SCOPE(LocEta0, Eta0);
+         OMEGA_SCOPE(LocKx, Kx);
+         OMEGA_SCOPE(LocKy, Ky);
+         OMEGA_SCOPE(LocAngFreq, AngFreq);
+
+         parallelFor(
+             {Mesh->NCellsAll, NVertLevels},
+             KOKKOS_LAMBDA(int ICell, int KLevel) {
+                R8 X     = LocXCell(ICell);
+                R8 Y     = LocYCell(ICell);
+                R8 Phase = LocKx * X + LocKy * Y - LocAngFreq * ElapsedTimeSec;
+                ThicknessTend(ICell, KLevel) +=
+                    LocEta0 *
+                    (-LocH0 * (LocKx + LocKy) * sin(Phase) -
+                     LocAngFreq * cos(Phase) +
+                     LocEta0 * (LocKx + LocKy) * cos(2.0_Real * Phase));
+             });
+      }
+   }; // end struct ManufacturedThicknessTendency
+
+   //===--------------------------------------------------------------------===/
+   // Manufactured tendency term for the momentum equation
+   //===--------------------------------------------------------------------===/
+   struct ManufacturedVelocityTendency {
+
+      // Constants defined in 'init'
+      TimeInstant ReferenceTime;
+      R8 Grav;
+      R8 Eta0;
+      R8 Kx;
+      R8 Ky;
+      R8 AngFreq;
+      R8 ViscDel2;
+      R8 ViscDel4;
+      bool VelDiffTendencyEnable;
+      bool VelHyperDiffTendencyEnable;
+
+      void operator()(Array2DReal NormalVelTend, const OceanState *State,
+                      const AuxiliaryState *AuxState, int ThickTimeLevel,
+                      int VelTimeLevel, TimeInstant Time) {
```

Comment:
Same as above, this function doesn't need to be defined in the header file. Its implementation can be moved to the cpp file. Also should probably have a `const` annotation:

```suggestion
      void operator()(Array2DReal NormalVelTend, const OceanState *State,
                      const AuxiliaryState *AuxState, int ThickTimeLevel,
                      int VelTimeLevel, TimeInstant Time) const {
```

Reply:
Done.